### ISSUE TITLE
DCX Library Components - Accessibility issues against all current components

### ISSUE THREAD

**Issue by ClaudioRicciCG** [Capgemini/dcx-react-library#200]
**Describe the bug**
Following an Accessibility Audit across all current Components, a list of bugs to be addressed has been compiled

**To Reproduce**
[Steps to reproduce the behavior:]
1. Install and enable webaim.org Chrome extension in Chrome
2. Go to localhost://1234
3. Click on a listed component
4. Click on the Webaim Icon top right of Chrome. 
5. View the Accessibility Warnings against the given component in the left hand panel.

**Expected behavior**
Once components have their Accessibility Issues fixed, the error count per Component in the Webaim extension should be zero.
**Listed Issues**
    <table>
        <thead>
            <tr><td><strong>COMPONENT</strong></td><td><strong>ISSUES</strong></td></tr>
        </thead>
        <tbody>
            <tr><td>radio</td><td colspan="4">Fieldset aria-describedby is empty. Needs to refer to a css selector of DOM element that's innerHTML would address the value</td></tr>
            <tr><td>Checkbox</td><td>Input type="checkbox" has  aria-describedby is empty. Needs to refer to a css selector of DOM element that's innerHTML would address the value.</td><td>Is data-aria-controls needed? As it is empty for each checkbox.</td><td>labels should refer using the 'for' attribute to the input it is for by the inputs id.</td><td>Fieldset aria-describedby is empty. Needs to refer to a css selector of DOM element that's innerHTML would address the value</td></tr>
            <tr><td>AutocompleteDemo</td><td colspan="4">2 Empty buttons need 'aria-label="something descriptive" in place</td></tr>
            <tr><td>ToggleDemo</td><td colspan="4">label is missing a htmlFor attribute, which should reference the immediate child div which should have an id</td></tr>
            <tr><td>FormDateDemo</td><td colspan="4">9 Empty buttons need 'aria-label="something descriptive" in place</td></tr>
            <tr><td>CopyToClipboardDemo</td><td colspan="4">3 Empty buttons need 'aria-label="something descriptive" in place</td></tr>
        </tbody>
    </table>`

**Desktop (please complete the following information):**
 - OS: Agnostic, any OS
 - Browser: Chrome (for testing, though affects all browser and devices)
 - Version: latest
